pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -271,4 +287,32 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(block_number: T::BlockNumber) -> Weight {
+			let active_season_id = Self::active_season_id();
+			let next_season_id = Self::next_active_season_id();
+			let season_id = active_season_id.unwrap_or(next_season_id);
+			let maybe_season = Self::seasons(season_id);
+
+			if let Some(season) = maybe_season {
+				if season.early_start <= block_number && block_number <= season.end {
+					ActiveSeasonId::<T>::put(season_id);
+					if block_number >= season.end {
+						NextActiveSeasonId::<T>::mutate(|season_id| season_id.saturating_inc());
+					}
+				} else {
+					ActiveSeasonId::<T>::kill();
+				}
+			}
```

Comment:
I think I made a mistake in the `NextActiveSeasonId` logic. We just need to put `season_id + 1` to just serve it as a lookup table when `ActiveSeasonId` is `None`:
```suggestion
			if let Some(season) = Self::seasons(season_id) {
				db_weight += T::DbWeight::get().reads(1);
				if season.early_start <= now && now <= season.end {
					ActiveSeasonId::<T>::put(season_id);
					NextActiveSeasonId::<T>::put(season_id.saturating_add(1));
					db_weight += T::DbWeight::get().writes(2);
				} else {
					ActiveSeasonId::<T>::kill();
					db_weight += T::DbWeight::get().writes(1);
				}
			}
			db_weight
```

Reply:
Much better this way, changed!